cumulus/client/executor-gossip/src/lib.rs

```diff
@@ -0,0 +1,302 @@
+#![allow(clippy::all)]
+
+mod worker;
+
+use self::worker::GossipWorker;
+use parity_scale_codec::{Decode, Encode};
+use parking_lot::{Mutex, RwLock};
+use sc_network::{ObservedRole, PeerId};
+use sc_network_gossip::{
+	GossipEngine, MessageIntent, Network as GossipNetwork, ValidationResult, Validator,
+	ValidatorContext,
+};
+use sc_utils::mpsc::TracingUnboundedReceiver;
+use sp_core::hashing::twox_64;
+use sp_executor::{Bundle, ExecutionReceipt};
+use sp_runtime::traits::{Block as BlockT, Hash as HashT, Header as HeaderT};
+use std::{
+	collections::HashSet,
+	fmt::Debug,
+	sync::Arc,
+	time::{Duration, Instant},
+};
+
+const LOG_TARGET: &str = "gossip::executor";
+
+const EXECUTOR_PROTOCOL_NAME: &str = "/subspace/executor/1";
+
+// TODO: proper timeout
+/// Timeout for rebroadcasting messages.
+/// The default value used in network-gossip is 1100ms.
+const REBROADCAST_AFTER: Duration = Duration::from_secs(6);
+
+type MessageHash = [u8; 8];
+
+/// Returns the configuration value to put in [`sc_network::config::NetworkConfiguration::extra_sets`].
+pub fn executor_gossip_peers_set_config() -> sc_network::config::NonDefaultSetConfig {
+	let mut cfg =
+		sc_network::config::NonDefaultSetConfig::new(EXECUTOR_PROTOCOL_NAME.into(), 1024 * 1024);
+	cfg.allow_non_reserved(25, 25);
+	cfg
+}
+
+/// Gossip engine messages topic.
+fn topic<Block: BlockT>() -> Block::Hash {
+	<<Block::Header as HeaderT>::Hashing as HashT>::hash(b"executor")
+}
```

Comment:
It only seems to be called in one place, so I'd just inline it

Reply:
It's indeed called once in `executor-gossip/src/lib.rs` but called a few times in `executor-gossip/src/worker.rs`.